pkg/sentry/devices/nvproxy/frontend.go

```diff
@@ -903,6 +903,20 @@ func ctrlGetNvU32List(fi *frontendIoctlState, ioctlParams *nvgpu.NVOS54_PARAMETE
 	return ctrlGetNvU32ListInvoke(fi, ioctlParams, &ctrlParams, list)
 }
 
+func ctrlBusGetInfo(fi *frontendIoctlState, ioctlParams *nvgpu.NVOS54_PARAMETERS) (uintptr, error) {
+	var ctrlParams nvgpu.NV2080_CTRL_BUS_GET_INFO_PARAMS
+	if ctrlParams.SizeBytes() != int(ioctlParams.ParamsSize) {
+		return 0, linuxerr.EINVAL
+	}
+	if _, err := ctrlParams.CopyIn(fi.t, addrFromP64(ioctlParams.Params)); err != nil {
+		return 0, err
+	}
+
+	busInfoBuf := make([]byte, ctrlParams.BusInfoListSize*nvgpu.CtrlXxxInfoSize)
```

Comment:
There needs to be a cap on `BusInfoListSize` to avoid a DoS vector from OOMing the Sentry here. Also need to check that `busInfoBuf` is non-empty. The user controls the size of this buffer and currently if I read the code correctly the user can set a size of zero.

The host kernel driver must also be doing similar checks; please verify the upper and lower thresholds it uses, and use the same thresholds here.